app/components/Story.js

```diff
@@ -2,34 +2,58 @@ import React, { Component } from 'react';
 import { StyleSheet, Text, View, TouchableHighlight, Dimensions,
  } from 'react-native';
 import StoryCard from './story/StoryCard';
+import StoryControlPaneContainer from '../containers/StoryControlPaneContainer';
 import ViewPager from 'react-native-viewpager';
 import Arabic from '../../stories/ar/chapters.js';
 import English from '../../stories/en/chapters.js';
 import Persian from '../../stories/fa/chapters.js';
 import Swedish from '../../stories/sv/chapters.js';
+import SideMenuContainer from '../containers/SideMenuContainer';
 
 class Story extends Component {
   constructor(props) {
     super(props);
 
-    const tempData = English.chapters[this.props.chapterId - 1].sections.map((section, index) =>
-      ({
-        topText: Arabic.chapters[this.props.chapterId - 1].sections[index],
-        bottomText: English.chapters[this.props.chapterId - 1].sections[index],
-      })
-    );
-
-    const ds = new ViewPager.DataSource({pageHasChanged: (r1, r2) => r1.text !== r2.text});
-    this.state = {
-      data: ds.cloneWithPages(tempData),
-      isSplit: this.props.viewMode === 'pair'
-    };
+    this.state = this._initializeDataSource();
 
     this.onClickBack = this.onClickBack.bind(this);
     this.onClickToggle = this.onClickToggle.bind(this);
     this._renderPage = this._renderPage.bind(this);
   }
 
+  _initializeDataSource(chapterId) {
```

Comment:
Ah, that's a convention I've used writing react code elsewhere.. Basically the underscore denotes that it's a "custom" component method, making it easy to tell at a glance custom functionality from the react lifecycle methods in a component. I can remove the underscores.